Sources/Crossroad/Router.swift

```diff
@@ -28,6 +28,11 @@ public final class Router<UserInfo> {
     }
 
     internal func register(_ route: Route<UserInfo>) {
+        if routes.contains(where: { element in
+            element.patternURL == route.patternURL
+        }) {
+            assertionFailure("\(route.patternURL) is already registered")
```

Comment:
`PatternURL` doesn't support pretty print.
Please build human-friendly string.

Reply:
`PatternURL.patternString` returns the registered String value as it is and use for this failure message. https://github.com/giginet/Crossroad/pull/32/commits/35e2e0622fb0fb0d33b8935895ddcf0d40664ed6